istioctl/cmd/analyze.go

```diff
@@ -137,6 +147,34 @@ istioctl analyze -L
 			sa := local.NewSourceAnalyzer(schema.MustGet(), analyzers.AllCombined(),
 				resource.Namespace(selectedNamespace), resource.Namespace(istioNamespace), nil, true)
 
+			// Check for suppressions and add them to our SourceAnalyzer
+			var suppressions []snapshotter.AnalysisSuppression
+			for _, s := range suppress {
+				parts := strings.Split(s, "=")
+				if len(parts) != 2 {
+					return fmt.Errorf("%s is not a valid suppression value. See istioctl analyze --help", s)
+				}
+				// Check to see if the supplied code is valid. If not, emit a
+				// warning but continue.
+				codeIsValid := false
+				for _, at := range msg.All() {
+					if at.Code() == parts[0] {
+						codeIsValid = true
+						break
+					}
+				}
+
+				if !codeIsValid {
+					fmt.Fprintf(cmd.ErrOrStderr(), "Warning: Supplied message code '%s' is an unknown message code and will be ignored.\n", parts[0])
```

Comment:
Maybe "will not be suppressed" to make the context clearer?

Reply:
Hmm, now that you point it out, I don't like what's there and am not super happy with the suggestion either.

Instead I updated the code so that we don't actually ignore the suppression if it's unknown, and just emit a warning saying it's unknown. What do you think?

Reply:
I agree that actually removing the suppression is redundant since it won't match anything anyway. My main concern here is that to the user, it may not be as obvious as it could be that 1) what they messed up was related to suppression and 2) what the impact is. 

I imagine most users will figure it out regardless but I'd rather word things to shorten that journey. :) 

Another possible alternate phrasing: "Warning: Suppressed message code '%s' is unknown and will be ignored."

Reply:
But if I'm adding it to the list of suppressions, technically it won't be ignored. We just expect it to not have any effect.

Reply:
So how about "Warning: Suppressed message code '%s' is unknown and will not have any effect." ?

Reply:
Done.